### PR TITLE
Add withdrawals

### PR DESCRIPTION
**Pull request by wborgeaud**
Add withdrawals as a public input array of `address, amount`. See EIP-4895.

### CODE REVIEW

**Comment by pgebheim** [comment]
Adding to RC1

File: evm/tests/withdrawals.rs [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+        &all_stark,
+        &[16..17, 10..11, 15..16, 14..15, 9..10, 12..13, 18..19], // Minimal ranges to prove an empty list
+        &config,
+    );
+
+    let mut timing = TimingTree::new("prove", log::Level::Info);
+    let (root_proof, _public_values) =
+        all_circuits.prove_root(&all_stark, &config, inputs, &mut timing)?;
+    timing.filter(Duration::from_millis(100)).print();
+    all_circuits.verify_root(root_proof.clone())
```
Comment (Nashtare) [inline evm/tests/withdrawals.rs:99]: Non-blocking, but is there any reason why we're using AllRecursiveCircuits here? And if so, wouldn't it deprecate `empty_txn_list`? As this one is run in the CI, we could combine the root_proof with a dummy one to still showcase `prove_aggreg` / `prove_block` in the CI.

**Review by Nashtare — APPROVED** [review]
Thanks @wborgeaud, looks good!

Just for my understanding, will we want at some point to include the `withdrawals_root` in the `BlockMetadata`? In which case we'd need to change the array layout to another MPT.

File: evm/tests/withdrawals.rs (hunk 1 above)
Comment (wborgeaud) [reply evm/tests/withdrawals.rs]: No reason really, I removed it.

**Comment by wborgeaud** [comment]
> Just for my understanding, will we want at some point to include the withdrawals_root in the BlockMetadata? In which case we'd need to change the array layout to another MPT.

Yes for sure! This is just a quick and dirty fix to be able to prove L1 blocks.

**Pull request merged**